perfdash/google-gcs-downloader.go

```diff
@@ -182,13 +182,21 @@ func getBuildData(result JobToCategoryData, prefix string, category string, labe
 	return result[prefix][category][label]
 }
 
-type bucketUtil struct {
+// BucketUtil is the interface that fetches data from a storage service
+type BucketUtil interface {
+	getBuildNumbersFromBucket(job string) ([]int, error)
+	listFilesInBuild(job string, buildNumber int, prefix string) ([]string, error)
+	getFileFromBucket(job string, buildNumber int, path string) ([]byte, error)
+}
+
+// GCSBucketUtil that creates a Google Cloud Storage client to fetch data
+type GCSBucketUtil struct {
 	client  *storage.Client
 	bucket  *storage.BucketHandle
 	logPath string
 }
 
-func newBucketUtil(bucket, path, credentialPath string) (*bucketUtil, error) {
+func newBucketUtil(bucket, path, credentialPath string) (BucketUtil, error) {
```

Comment:
Probably want to name this `NewGCSBucketUtil`, but the signature (inputs and outputs) look good!